syft/pkg/cataloger/cataloger.go

```diff
@@ -126,14 +133,32 @@ func RequestedAllCatalogers(cfg Config) bool {
 	return false
 }
 
-func filterCatalogers(catalogers []Cataloger, enabledCatalogerPatterns []string) []Cataloger {
+func filterForExternalSources(catalogers []Cataloger, cfg Config) []Cataloger {
```

Comment:
i feel like it should be up front with the user that it was not able to run because there's a configuration conflict than silently dropping catalogers - may be deceiving to the user.